diff -ruN xserver-20060318./xorg/GL/symlink-mesa.sh xserver-20060318/xorg/GL/symlink-mesa.sh
--- xserver-20060318./xorg/GL/symlink-mesa.sh	2006-03-08 19:32:31.000000000 +0100
+++ xserver-20060318/xorg/GL/symlink-mesa.sh	2006-03-18 15:11:07.287819064 +0100
@@ -412,6 +412,8 @@
     src_dir src/mesa/shader/slang
     dst_dir mesa/shader/slang
 
+    action slang_analyse.c
+    action slang_analyse.h
     action slang_assemble.c
     action slang_assemble.h
     action slang_assemble_assignment.c
@@ -438,6 +440,8 @@
     action slang_export.h
     action slang_library_noise.c
     action slang_library_noise.h
+    action slang_library_texsample.c
+    action slang_library_texsample.h
     action slang_link.c
     action slang_link.h
     action slang_mesa.h
--- a/xorg/GL/mesa/shader/slang/Makefile.am	2006-03-16 15:45:13.204195192 -0500
+++ a/xorg/GL/mesa/shader/slang/Makefile.am.new	2006-03-16 15:52:56.529759000 -0500
@@ -17,6 +17,7 @@
            -I$(top_srcdir)/hw/xfree86/os-support
 
 nodist_libslang_la_SOURCES = slang_assemble_assignment.c \
+		      slang_analyse.c \
 		      slang_assemble.c \
 		      slang_assemble_conditional.c \
 		      slang_assemble_constructor.c \
@@ -29,6 +30,7 @@
 		      slang_execute.c \
 		      slang_export.c \
 		      slang_library_noise.c \
+		      slang_library_texsample.c \
 		      slang_link.c \
 		      slang_preprocess.c \
 		      slang_storage.c \
